playwright/_impl/_connection.py

```diff
@@ -294,6 +299,19 @@ def dispatch(self, msg: ParsedMessagePayload) -> None:
                 parent, params["type"], params["guid"], params["initializer"]
             )
             return
+
+        object = self._objects.get(guid)
+        if not object:
+            raise Exception(f'Cannot find object to "{method}": {guid}')
+
+        if method == "__adopt__":
+            child_guid = cast(Dict[str, str], params)["guid"]
+            child = self._objects.get(child_guid)
```

Comment:
```suggestion
            child = self._objects.get(guid)
```

why not that?

Reply:
It's a different guid. `child_guid` comes from `params` while `guid` comes from `msg` (`guid = msg["guid"]`)